amaranth/lib/wiring.py

```diff
@@ -1574,6 +1583,12 @@ def connect_dimensions(dimensions, *, out_path, in_path):
                                        out_path=(*out_path, index), in_path=(*in_path, index))
             assert out_member.dimensions == in_member.dimensions
             connect_dimensions(out_member.dimensions, out_path=out_path, in_path=in_path)
+
+    # If no connections were made, and there were inputs but no outputs in the
+    # signatures, issue a diagnostic as this is most likely in error.
+    if len(connections) == 0 and any_in and not any_out:
+        raise ConnectionError(f"Cannot connect without at least one output")
```

Comment:
I propose an improved wording, something like this (edits welcome):

```suggestion
        raise ConnectionError(f"Only input to input connections have been made between several "
                              f"interfaces; should one of them have been flipped?")
```

Reply:
Thanks, much better! Adopted, tests adjusted & squashed.